### PR TITLE
Use deferred renderer on Android

### PR DESCRIPTION
**Pull request by jp2masa**
## What does the pull request do?

- Use deferred renderer on Android.
- Implemented render timer based on Choreographer.
- Removed apparently useless `ActivityTracker`.
- ~Removed useless EGL classes for Android that I added in the last PR, as there's a common implementation that works perfectly but I hadn't seen it.~
- Handle surface changes on OpenGL.

## What is the current behavior?

Only immediate renderer works on Android.

## What is the updated/expected behavior with this PR?

It is now possible to use deferred renderer on Android, and it's set as default. Performance is much better now. ~Unfortunately, `OnPause`/`OnResume` isn't working, it shows a black screen, and I don't know if it was like that before, I'll take a look at that soon.~

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

**Comment by jp2masa** [comment]
The `OnPause`/`OnResume` problem is because the OpenGL surface has to be marked as corrupted when surface changes, the fix is simple so I will add it to this PR (also need to revert the deleted EGL classes as they are now needed).

**Comment by grokys** [comment]
@jp2masa slightly off topic, but I wonder if https://github.com/AvaloniaUI/Avalonia/pull/5070 improves performance in Android noticeably? If you're working on Android at the moment I'd be interested to know (or if it causes problems).

**Comment by jp2masa** [comment]
It doesn't seem to make much difference, I believe the problem isn't related to styles (switching styles isn't much slower than Windows as well for example). I actually profiled it a few days ago and I think the problem is that it spends more than 1 min for syscalls when loading the app, probably related to dynamic loading and the native part in general.

**Comment by Gillibald** [comment]
Is this with a release build without debugger etc?

**Comment by jp2masa** [comment]
For profiling it's required to have debugging enabled, so it's slower, but I already tried disabling it and enabling AOT and although it's faster, it still takes more than 30s to load IIRC.

**Comment by Gillibald** [comment]
I wonder how well Xamarin.Forms and Uno perform in that area. We definitely need some benchmarks for startup etc.

File: src/Android/Avalonia.Android/AvaloniaActivity.cs [hunk 1]
```diff
@@ -1,27 +1,50 @@
...
             View = new AvaloniaView(this);
-            if(_content != null)
+            if (_content != null)
                 View.Content = _content;
             SetContentView(View);
             TakeKeyEvents(true);
             base.OnCreate(savedInstanceState);
         }
 
+        protected override void OnResume()
```
Comment (kekekeks) [inline src/Android/Avalonia.Android/AvaloniaActivity.cs:24]: That will require an Avalonia-based Activity which prevents AvaloniaView from being used for embedding scenarios

Comment (jp2masa) [reply src/Android/Avalonia.Android/AvaloniaActivity.cs]: I updated the PR to replace this with manual `Tick` subscription count. However, I noticed it isn't enough to stop the timer as the animation clock ([`Avalonia.Animation.RenderLoopClock`](https://github.com/AvaloniaUI/Avalonia/blob/6ff478bdb131b9643d3d3bb17caf50c5bcba3181/src/Avalonia.Visuals/Animation/RenderLoopClock.cs)) never stops, so the render loop doesn't stop, i.e. `_items.Count` never reaches 0 here:

https://github.com/AvaloniaUI/Avalonia/blob/6ff478bdb131b9643d3d3bb17caf50c5bcba3181/src/Avalonia.Visuals/Rendering/RenderLoop.cs#L87-L90

Is this a bug? Or am I missing something?

Comment (kekekeks) [reply src/Android/Avalonia.Android/AvaloniaActivity.cs]: It's a known issue. For now I'd suggest to manually cast IRenderTimer to a concrete type and manually notify it about the subscription count

**Comment by ili** [comment]
@jp2masa @kekekeks what's about this PR?
It shows significant ~prefomance~ application response time improvement.

**Review by kekekeks — APPROVED** [review]

**Pull request merged**